mlir/lib/Dialect/Rock/Transforms/FoldBroadcast.cpp

```diff
@@ -74,8 +74,8 @@ struct FoldBroadcast : public OpRewritePattern<rock::GemmOp> {
             // anything, otherwise we need to be sure that the new
             // dimension is a single-length broadcast
             case rock::TransformType::Broadcast:
-              if (tr.getParams().back() != 1)
-                newWorkList.insert(tr.getLowerDims().back());
+              if (tr.getParams()[idx] != 1)
+                newWorkList.insert(tr.getLowerDims()[idx]);
```

Comment:
Ok orthogonal to this fix, I'm curious of how the original case condition works. Do we have any broadcast that falls into this if condition? My assumption is that `tr.getParams()[idx] == 1` should always satisfy because broadcast only support from unit dimension.

Reply:
I designed Broadcast to support the general case, and while I don't think we currently have any clients that call for that, it's - IMO - a very reasonable thing to support